commands-yml/parse.js

```diff
@@ -0,0 +1,87 @@
+/* eslint-disable no-console */
+import path from 'path';
+import yaml from 'yaml-js';
+import { fs } from 'appium-support';
+import validate from 'validate.js';
+import Handlebars from 'handlebars';
+import replaceExt from 'replace-ext';
+import validator from './validator';
+
+// Create Handlebars helper that shows a version range
+Handlebars.registerHelper('versions', (object, name) => {
+  if (!object) {
+    return 'None';
+  }
+
+  if (object === true) {
+    return 'All';
+  }
+
+  const min = object[name ? `${name}_min` : 'min'];
+  const max = object[name ? `${name}_max` : 'max'];
+
+  if (!min && !max) {
+    return 'All';
+  } else if (!max) {
+    return `${min}+`;
+  } else if (!min) {
+    return `<= ${max}`;
+  } else {
+    return `${min} to ${max}`;
+  }
+});
+
+Handlebars.registerHelper('hyphenate', (str) => {
+  return str.replace('_', '-');
+});
+
+Handlebars.registerHelper('uppercase', (str) => {
+  return str.toUpperCase();
+});
+
+Handlebars.registerHelper('capitalize', (driverName) => {
+  switch (driverName) {
+    case 'xcuitest':
+      return 'XCUITest';
+    case 'uiautomation':
+      return 'UIAutomation';
+    case 'uiautomator2':
+      return 'UiAutomator2';
+    case 'uiautomator':
+      return 'UiAutomator';
+    default:
+      return driverName[0].toUpperCase() + driverName.substr(1);
+  }
+});
+
+(async function () {
```

Comment:
rather than lose the whitespace due to the try/catch, you could just do this:

```js
async function main ( ) {
  // all your code, but not in a try catch
}

asyncify(main); // asyncbox's asyncify takes care of error logging
```